src/Azure.Functions.ArtifactAssembler/ArtifactAssembler.cs

```diff
@@ -161,12 +198,150 @@ private async Task CreateVisualStudioCoreToolsAsync()
                 Directory.Delete(consolidatedArtifactDirPath, true);
             });
 
+            await Task.WhenAll(packTasks);
+            Console.WriteLine($"Finished assembling Visual Studio Core Tools artifacts");
+        }
+
+        private async Task CreateCliCoreToolsAsync()
+        {
+            Console.WriteLine($"Starting to assemble CLI Core Tools artifacts");
+
+            // Create a directory to store the assembled artifacts.
+            var cliCoreToolsTargetArtifactDir = Path.Combine(_stagingDirectory, CliOutputArtifactDirectoryName);
+            Directory.CreateDirectory(cliCoreToolsTargetArtifactDir);
+            string outOfProcVersion = string.Empty;
+            string inProcVersion = string.Empty;
+            string outOfProcArtifactDirPath = string.Empty;
+            string inProc8ArtifactDirPath = string.Empty;
+
+            var packTasks = _cliArtifacts.Select(async artifactName =>
+            {
+                // If we are running this for the first time, extract the directory path and out of proc version
+                if (String.IsNullOrEmpty(outOfProcArtifactDirPath))
+                {
+                    var outOfProcResponse = GetArtifactDirectoryAndVersionNumber(_outOfProcExtractedRootDir, artifactName);
+                    outOfProcArtifactDirPath = outOfProcResponse[0];
+                    outOfProcVersion = outOfProcResponse[1];
+                }
+                else
+                {
+                    var artifactNameWithVersion = $"{artifactName}.{outOfProcVersion}";
+                    outOfProcArtifactDirPath = Path.Combine(_outOfProcExtractedRootDir, artifactNameWithVersion);
+                }
+
+                // Create a new directory to store the oop core tools with in-proc8 and in-proc6 files.
+                var outOfProcArtifactName = Path.GetFileName(outOfProcArtifactDirPath);
```

Comment:
wonder we could move this to a small method which can be called from here and from `CreateVisualStudioCoreToolsAsync` method?

Reply:
the logic is a bit different so I decided to separate them to keep it easier to parse/read